fastfood/food.py

```diff
@@ -42,34 +41,31 @@ def build_cookbook(build_config, templatepack, cookbooks_home,
     with open(build_config) as cfg:
         cfg = json.load(cfg)
 
-    cookbook = None
-    if cookbook_path:
-        try:
-            cookbook = book.CookBook(cookbook_path)
-        # change this to a more specific error
-        except ValueError:
-            # create a new one please
-            pass
+    cookbook_name = cfg['name']
+    cookbook_path = os.path.join(cookbooks_home, cookbook_name)
 
     written_files = []
-    if not cookbook:
-        cookbook_name = cfg['name']
-        files, cookbook = create_new_cookbook(
-            cookbook_name, templatepack, cookbooks_home, force=force)
-        written_files.extend(files)
+    cookbook = create_new_cookbook(
+        cookbook_name, cookbooks_home, force=force)
+    cookbook_path = cookbook.path
 
     for stencil in cfg['stencils']:
         stencil_set = stencil.pop('stencil_set')
         # items left un-popped are **options
-        files, updated_cookbook = update_cookbook(
-            cookbook, templatepack, stencil_set, **stencil)
+        files, updated_cookbook = process_stencil(
+            cookbook_name,
+            cookbook_path,
+            templatepack,
+            stencil_set,
+            **stencil
+            )
         written_files.extend(files)
-        assert updated_cookbook is cookbook
     return written_files, cookbook
 
 
-def update_cookbook(cookbook, templatepack, stencilset_name, **options):
-    """Update the cookbook with the stencil + options.
+def process_stencil(cookbook_name, cookbook_path, templatepack,
```

Comment:
I was able to get away with only passing `cookbook_name` to `process_stencil`. But we'll need the name since there isn't one stored in a `Cookbook` instance (the class relies on metadata which may or may not exist).